### PR TITLE
Add static TrialResult.from_single_parameter_set factory method

### PR DESCRIPTION
**Pull request by Strilanc**
- The goal here is to gradually refactor TrialResult into something that can also represent sweep results.

### CODE REVIEW

File: cirq/study/trial_result.py [hunk 1]
```diff
@@ -135,6 +135,23 @@ def __init__(
             converted_dict[key] = [pd.Series(m_vals) for m_vals in val]
         self.data = pd.DataFrame(converted_dict)
 
+    @staticmethod
+    def from_single_parameter_set_reps(
```
Comment (cduck) [inline cirq/study/trial_result.py]: Nit: Maybe just `from_single_parameter_set`.

File: cirq/study/trial_result.py [hunk 2]
```diff
@@ -135,6 +135,23 @@ def __init__(
...
+            *,  # Forces keyword args.
+            params: resolver.ParamResolver,
+            measurements: Dict[str, np.ndarray]) -> None:
```
Comment (cduck) [inline cirq/study/trial_result.py]: Return type.

File: cirq/study/trial_result_test.py [hunk 3]
```diff
@@ -20,39 +20,38 @@
...
-    v = cirq.TrialResult(params=cirq.ParamResolver({'a': 2}),
-                         measurements={'xy': np.array([[1, 0], [0, 1]])})
-
-    assert repr(v) == ("cirq.TrialResult(params=cirq.ParamResolver({'a': 2}), "
-                       "repetitions=2, measurements={'xy': array([[1, 0],\n"
-                       "       [0, 1]])})")
+    v = cirq.TrialResult.from_single_parameter_set_reps(
+        params=cirq.ParamResolver({'a': 2}),
+        measurements={'xy': np.array([[1, 0], [0, 1]])})
+    cirq.testing.assert_equivalent_repr(v)
```
Comment (cduck) [inline cirq/study/trial_result_test.py]: Needs `cirq` and `array` in global_vals.

Comment (Strilanc) [reply cirq/study/trial_result_test.py]: Fixed in a different way.

File: cirq/study/trial_result.py (hunk 2 above)
Comment (Strilanc) [reply cirq/study/trial_result.py]: Done.

File: cirq/study/trial_result.py (hunk 1 above)
Comment (Strilanc) [reply cirq/study/trial_result.py]: Done.

File: cirq/study/trial_result.py [hunk 4]
```diff
@@ -266,11 +267,17 @@ def histogram(  # type: ignore
...
+            key, val = entry
+            return '{!r}: {}'.format(key, proper_repr(val))
+
+        measurement_dict_repr = (
+                '{' +
+                ', '.join([item_repr(e) for e in self.measurements.items()]) +
+                '}')
+
+        return 'cirq.TrialResult(params={!r}, measurements={})'.format(
+            self.params, measurement_dict_repr)
```
Comment (cduck) [inline cirq/study/trial_result.py]: What does this new repr look like?

Comment (Strilanc) [reply cirq/study/trial_result.py]:
```
cirq.TrialResult(params=cirq.ParamResolver({'a': 2}), measurements={'xy': np.array([[1, 0], [0, 1]])})
```

Comment (Strilanc) [reply cirq/study/trial_result.py]: The main criteria of a repr is that it round trips, so I'm not too worried about the details. It happens to look pretty good.

**Pull request merged**